src/main/java/net/obvj/performetrics/Stopwatch.java

```diff
@@ -241,37 +239,55 @@ public boolean isStarted()
     }
 
     /**
-     * Returns the counters associated with this stopwatch instance.
+     * Returns the counter types associated with this stopwatch instance.
+     *
+     * @return all counter types associated with this stopwatch instance
+     */
+    public List<Type> getTypes()
+    {
+        return types;
+    }
+
+    /**
+     * Returns a list of counters populated by this stopwatch.
+     * <p>
+     * New counters are created every time the {@code start()} method is called.
      *
      * @return all counters associated with this stopwatch instance
      */
-    public Collection<Counter> getCounters()
+    public List<Counter> getCounters()
     {
-        return counters.values();
+        return sessions.stream().map(TimingSession::getCounters).flatMap(Collection::stream)
+                .collect(Collectors.toList());
     }
 
     /**
-     * Returns the counter instance associated with a given type in this stopwatch.
+     * Returns a list of counters associated with a given type in this stopwatch.
      *
      * @param type the counter type to be fetched
-     * @return the counter instance associated with the given type in this stopwatch
-     * @throws IllegalArgumentException if the specified type is not available in this
-     *                                  stopwatch instance
+     * @return a list of populated counters for the specified type, or an empty list if no
+     *         counters found
+     * @since 2.2.0
      */
-    public Counter getCounter(Type type)
+    public List<Counter> getCounters(Type type)
     {
-        if (!counters.containsKey(type))
-        {
-            throw new IllegalArgumentException(
-                    MessageFormat.format(MSG_TYPE_NOT_AVAILABLE, type, counters.keySet()));
-        }

```

Comment:
Move this portion of code to `getCountersAsStream()` to avoid duplication in other methods.